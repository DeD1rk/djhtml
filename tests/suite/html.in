<!-- Basic HTML sanity check -->
<!doctype html>
<html>
<head>
<title>Plain</title>
<script src="test.js"></script>
<link rel="stylesheet" href="test.css">
</head>
<body>
<h1>Title</h1>
</body>
</html>


<!-- Multiple closing/opening tags on same line -->
<div>
    text
</div><div>
    text
</div>


<!-- Tag continuation -->
<div
id="id"
class="class">
text
</div>


<!-- Ignore self-closing tags -->
<div>
<div/>
text
</div>


<!-- Ignore void tags -->
<img>
<input>
<input />
<input></input>
<!doctype>


<!-- Don't parse style/script as html -->
<script>
<ul>
<li>
</script>

<style>
<style>
</style>


<!-- Except for JS templates... -->
<script id="id" type="text/template" class="class">
<ul>
<li/>
</ul>
</script>


<!-- Ignore multiline comments -->
<!--
 1
  2
   3
-->


<!-- Ignore <pre> elements -->
<pre>
   <dl>
  <dt>
 <dd>
</pre>


<!-- Remove trailing spaces -->
Invisible spaces here:   


<!-- Append final newline -->