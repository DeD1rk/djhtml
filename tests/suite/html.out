<!-- Basic HTML sanity check -->
<!doctype html>
<html>
    <head>
        <title>Plain</title>
        <script src="test.js"></script>
        <link rel="stylesheet" href="test.css">
    </head>
    <body>
        <h1>Title</h1>
    </body>
</html>


<!-- Multiple closing/opening tags on same line -->
<div>
    text
</div><div>
    text
</div>


<!-- Some crazy edgecases that even the mighty web-mode fails to indent correctly -->

<div>
    <div>
        <div>
            text
        </div></div></div>

<div><div><div>
    text
</div>
</div>
</div>

<div0><div1>
    text2</div1><div1><div2>
        text3
    </div2><div2>
        text3
    </div2>
    </div1>
</div0>

<div0>
    <div1>
        <div2>
            text3
        </div2></div1></div0><div0>
            text1
        </div0>


<!-- Tag continuation -->
<div
    id="id"
    class="class">
    text
</div>

<div
    id="id"
    class="class"
>
    text
</div>


<!-- Ignore self-closing tags -->
<div>
    <div/>
    text
</div>


<!-- Ignore void tags -->
<img>
<input>
<input />
<input></input>
<!doctype>


<!-- Don't parse style/script as html -->
<script>
    <ul>
    <li>
</script>

<style>
    <style>
</style>


<!-- Except for JS templates... -->
<script id="id" type="text/template" class="class">
    <ul>
        <li/>
    </ul>
</script>


<!-- Ignore multiline comments -->
<!--
 1
  2
   3
-->


<!-- Ignore <pre> elements -->
<pre>
   <dl>
  <dt>
 <dd>
</pre>


<!-- Real life example kindly provided by @jacklinke -->
<div class="row">
    <div class="input-group date" data-provide="datepicker" data-date-format="yyyy/mm/dd">
        <div class="btn">
        </div></div>
</div><div>
    <div class="custom-control custom-switch">
        <input type="checkbox" name="filter-changes" id="filter-changes" class="custom-control-input fs-nano" value="0" {% if request.session.manage_filter_changes %}{% if request.session.manage_filter_changes == True %}checked{% endif %}{% endif %}>
    </div>
    <form class="form-inline">
        <div class="custom-control custom-switch">
            <input type="checkbox" name="filter-status-new" id="filter-status-new" class="custom-control-input fs-nano" value="0" {% if request.session.manage_filter_status_new %}{% if request.session.manage_filter_status_new == True %}checked{% endif %}{% endif %}>
        </div>
    </form>
</div>


<!-- Multiline elements -->
<img src="/some/long/path"
    id="1">
<blockquote class="dishy"
    id="2" />
<a href="/some/long/path"
    id="3"></a>

<!-- Multiline attribute values -->
<svg width="325" height="325" xmlns="http://www.w3.org/2000/svg">
    <path
        d="M 80 80
            A 45 45, 0, 0, 0, 125 125
            L 125 80 Z"
        fill="green"
    />
    <path
        d="
            M 80 80
            A 45 45, 0, 0, 0, 125 125
            L 125 80 Z
        "
        fill="green"
    />
    <path d="M 80 80
        A 45 45, 0, 0, 0, 125 125
        L 125 80 Z"
        fill="green"
    />
</svg>


<!-- Remove trailing spaces -->
Invisible spaces here:


<!-- Do not append final newline -->